source/extensions/tracers/xray/config.cc

```diff
@@ -0,0 +1,47 @@
+#include "extensions/tracers/xray/config.h"
+
+#include <string>
+
+#include "envoy/registry/registry.h"
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/well_known_names.h"
+#include "extensions/tracers/xray/xray_tracer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+XRayTracerFactory::XRayTracerFactory() : FactoryBase(TracerNames::get().XRay) {}
+
+Tracing::HttpTracerPtr
+XRayTracerFactory::createHttpTracerTyped(const envoy::config::trace::v2::XRayConfig& proto_config,
+                                         Server::Instance& server) {
+  std::string sampling_rules_json;
+  try {
+    sampling_rules_json =
+        server.api().fileSystem().fileReadToEnd(proto_config.sampling_rule_manifest().filename());
```

Comment:
Please use the standard `DataSource` libraries here to read, they will take care of the inline case.

Reply:
Is there an example in the source code that I can look at? (new to protobufs)

Reply:
@marcomagdy

Its just a wrapper for what you're already doing. But I believe its here: https://github.com/envoyproxy/envoy/blob/88fe0c9310d67fd4c8797dc37278641dd9415fd0/source/common/config/datasource.cc#L9

Reply:
Yep, thaks @lavignes.